### PR TITLE
Just the protobuf parts of #905

### PR DESCRIPTION
**Pull request by ekzhang**
As described in [the main PR](https://github.com/modal-labs/modal/pull/7745), this needs to be merged first.

image

### CODE REVIEW

**Review by erikbern — APPROVED** [review]
LGTM

Mentioned it in https://github.com/modal-labs/modal-client/pull/905#discussion_r1336019072 – but would it make sense to add some unit tests for this (i.e. the ASGI serialization in itself)? Not sure how easy/hard that would be. Not a blocker for merging this but a minor preference to do if it's easy.

File: modal_utils/rand_pb_testing.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+def rand_pb(proto: Type[T], rand: Optional[Random] = None) -> T:
+    """Generate a pseudorandom protobuf message.
+
+    **Example:**
+
+    ```python
+    r = random.
+    definition = rand_pb(api_pb2.Function, random)
+    ```
+    """
```
Comment (ekzhang) [inline modal_utils/rand_pb_testing.py:85]: I wrote this function to help with the serialization tests

File: modal_utils/rand_pb_testing.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+def rand_pb(proto: Type[T], rand: Optional[Random] = None) -> T:
+    """Generate a pseudorandom protobuf message.
+
+    ```python
+    rand = random.Random(42)
+    definition = rand_pb(api_pb2.Function, rand)
+    ```
+    """
```
Comment (ekzhang) [inline modal_utils/rand_pb_testing.py:83]: I wrote this function to help with property-based testing of the ASGI serialization

Comment (ekzhang) [reply modal_utils/rand_pb_testing.py:83]: Feel free to use it in other plaes though, should work with any protobuf

**Pull request merged**